### PR TITLE
Use signatures in Schema::Result::Needles

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Needles.pm [hunk 1]
```diff
@@ -75,26 +75,17 @@ __PACKAGE__->belongs_to(
...
 }
 
-sub update_needle_cache {
-    my ($needle_cache) = @_;
-
-    for my $needle (values %$needle_cache) {
-        $needle->update;
-    }
-}
+sub update_needle_cache ($needle_cache) { $_->update for (%$needle_cache) }
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/Needles.pm:83]:
```suggestion
sub update_needle_cache ($needle_cache) { $_->update for values %$needle_cache }
```

File: lib/OpenQA/Schema/Result/Needles.pm [hunk 2]
```diff
@@ -203,15 +189,9 @@ sub remove {
...
-    my ($self) = @_;
-
-    return $self->file_present(-e $self->path ? 1 : 0);
-}
-
-sub to_json {
-    my ($self, $controller) = @_;
+sub check_file ($self) { $self->file_present(-e $self->path ? 1 : 0) }
 
+sub to_json ($self, $controller) {
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/Needles.pm:194]: Looks like `$controller` isn't used. Maybe it makes more sense to removed (removing it from the call site as well).

Comment (perlpunk) [reply lib/OpenQA/Schema/Result/Needles.pm:194]: Yeah, this looks like an accident. It is never called with a second argument, so just remove it.

Comment (okurz) [reply lib/OpenQA/Schema/Result/Needles.pm]: done

File: lib/OpenQA/Schema/Result/Needles.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Schema/Result/Needles.pm]: done

**Review by perlpunk — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**